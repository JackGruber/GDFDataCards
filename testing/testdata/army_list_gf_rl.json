{
    "id": "ocjUa4aH72ba",
    "gameSystem": "gf",
    "armyId": "4k5amkxoybdiqotm",
    "armyIds": [
        "4k5amkxoybdiqotm"
    ],
    "armyFaction": null,
    "armyName": "Robot Legions",
    "modified": "2024-03-18T15:35:23.910Z",
    "saveVersion": 3,
    "favourite": false,
    "armyVersions": [
        {
            "version": "3.2.0",
            "armyId": "4k5amkxoybdiqotm"
        }
    ],
    "listPoints": 1205,
    "list": {
        "id": "ocjUa4aH72ba",
        "name": "Necrons Combat Patrol 1.2k",
        "points": 1200,
        "isCloud": true,
        "modified": "2024-03-18T15:35:23.910Z",
        "gameSystem": "gf",
        "competitive": true,
        "pointsLimit": 1200,
        "unitPreview": null,
        "campaignMode": false,
        "selectedUnitId": "FnPS0",
        "undoUnitRemove": [
            {
                "id": "wvgi7_R",
                "xp": 0,
                "cost": 175,
                "name": "Eternals",
                "size": 5,
                "notes": null,
                "valid": true,
                "armyId": "4k5amkxoybdiqotm",
                "sortId": 6,
                "traits": [],
                "defense": 2,
                "loadout": [
                    {
                        "id": "_WpwGRUk",
                        "name": "CCW",
                        "type": "ArmyBookWeapon",
                        "count": 5,
                        "label": "CCW (A1)",
                        "range": 0,
                        "attacks": 1,
                        "weaponId": "4o5CBBZC",
                        "specialRules": [],
                        "originalCount": 5
                    },
                    {
                        "id": "dT086NsK",
                        "name": "Flux Carbine",
                        "type": "ArmyBookWeapon",
                        "count": 5,
                        "label": "Flux Carbine (18\", A2, Flux)",
                        "range": 18,
                        "attacks": 2,
                        "weaponId": "dT086NsK",
                        "specialRules": [
                            {
                                "key": "flux",
                                "name": "Flux",
                                "modify": false,
                                "rating": ""
                            }
                        ],
                        "originalCount": 5
                    }
                ],
                "quality": 3,
                "combined": false,
                "upgrades": [
                    "F1"
                ],
                "equipment": [
                    {
                        "id": "_WpwGRUk",
                        "name": "CCW",
                        "type": "ArmyBookWeapon",
                        "count": 5,
                        "label": "CCW (A1)",
                        "range": 0,
                        "attacks": 1,
                        "weaponId": "4o5CBBZC",
                        "specialRules": [],
                        "originalCount": 5
                    },
                    {
                        "id": "dT086NsK",
                        "name": "Flux Carbine",
                        "type": "ArmyBookWeapon",
                        "count": 5,
                        "label": "Flux Carbine (18\", A2, Flux)",
                        "range": 18,
                        "attacks": 2,
                        "weaponId": "dT086NsK",
                        "specialRules": [
                            {
                                "key": "flux",
                                "name": "Flux",
                                "modify": false,
                                "rating": ""
                            }
                        ],
                        "originalCount": 5
                    }
                ],
                "joinToUnit": null,
                "selectionId": "_xivy",
                "specialRules": [
                    {
                        "key": "regeneration",
                        "name": "Regeneration",
                        "rating": ""
                    },
                    {
                        "key": "robot",
                        "name": "Robot",
                        "rating": ""
                    },
                    {
                        "key": "slow",
                        "name": "Slow",
                        "rating": ""
                    }
                ],
                "disabledSections": [],
                "selectedUpgrades": [],
                "disabledUpgradeSections": []
            }
        ],
        "activationCount": 5,
        "modelCount": 15,
        "units": [
            {
                "id": "y8xIlHV",
                "armyId": "4k5amkxoybdiqotm",
                "customName": "Hochlord Oltyx, mit Tachyonenpfeil",
                "selectionId": "FnPS0",
                "selectedUpgrades": [
                    {
                        "instanceId": "rh9wbmb8z",
                        "upgradeId": "IxcS2nP",
                        "optionId": "pAgesdx"
                    },
                    {
                        "instanceId": "0FSe9XBLo",
                        "upgradeId": "nl8X7bN",
                        "optionId": "cYskqiN"
                    },
                    {
                        "instanceId": "yxEhMJFlF",
                        "upgradeId": "i2wAxsw",
                        "optionId": "uiYOefY"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "IMHCbyw",
                "armyId": "4k5amkxoybdiqotm",
                "customName": "Skorpekh-Destruktoren",
                "selectionId": "Kz7AE",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "xRIfeJR",
                "armyId": "4k5amkxoybdiqotm",
                "customName": "Kanoptech-Dominatorschreiter",
                "selectionId": "RsN0j",
                "selectedUpgrades": [],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "njiIkJi",
                "armyId": "4k5amkxoybdiqotm",
                "customName": "Necronkrieger",
                "selectionId": "Q8tjV",
                "selectedUpgrades": [
                    {
                        "instanceId": "xcc6NF4gJ",
                        "upgradeId": "sPruWe5",
                        "optionId": "r_yHp"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            },
            {
                "id": "njiIkJi",
                "armyId": "4k5amkxoybdiqotm",
                "customName": "Necronkrieger",
                "selectionId": "9eu-0",
                "selectedUpgrades": [
                    {
                        "instanceId": "kXlHsiN8x",
                        "upgradeId": "sPruWe5",
                        "optionId": "r_yHp"
                    }
                ],
                "combined": false,
                "joinToUnit": null,
                "xp": 0,
                "traits": [],
                "notes": null
            }
        ]
    }
}