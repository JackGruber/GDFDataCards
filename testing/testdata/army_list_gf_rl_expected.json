{
    "armyId": "4k5amkxoybdiqotm",
    "gameSystem": "gf",
    "gameSystemId": 2,
    "armyName": "Robot Legions",
    "listPoints": 1205,
    "listName": "Necrons Combat Patrol 1.2k",
    "units": [
        {
            "type": "Annihilator Lord",
            "name": "Hochlord Oltyx, mit Tachyonenpfeil",
            "armyId": "4k5amkxoybdiqotm",
            "id": "y8xIlHV",
            "cost": 150,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "A1",
                "C1",
                "D1"
            ],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 1,
                    "name": "Doom Scythe",
                    "specialRules": [
                        {
                            "name": "Deadly",
                            "rating": 3,
                            "label": "Deadly(3)"
                        }
                    ],
                    "ap": 2
                },
                {
                    "attacks": 3,
                    "count": 1,
                    "name": "Wrist-Laser",
                    "range": 18,
                    "specialRules": [],
                    "ap": "3"
                }
            ],
            "specialRules": [
                {
                    "key": "ambush",
                    "name": "Ambush",
                    "rating": "",
                    "label": "Ambush"
                },
                {
                    "key": "hero",
                    "name": "Hero",
                    "rating": "",
                    "label": "Hero"
                },
                {
                    "key": "robot",
                    "name": "Robot",
                    "rating": "",
                    "label": "Robot"
                },
                {
                    "id": 964,
                    "key": "self-repair",
                    "name": "Self-Repair",
                    "label": "Self-Repair"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "6",
                    "label": "Tough(6)"
                }
            ],
            "upgradeCost": [
                40,
                10,
                35
            ],
            "equipment": [
                {
                    "name": "Technomancer",
                    "specialRules": [
                        {
                            "key": "caster",
                            "name": "Caster",
                            "type": "ArmyBookRule",
                            "label": "Caster(2)",
                            "rating": 2
                        }
                    ]
                }
            ]
        },
        {
            "type": "Annihilators",
            "name": "Skorpekh-Destruktoren",
            "armyId": "4k5amkxoybdiqotm",
            "id": "IMHCbyw",
            "cost": 280,
            "defense": 3,
            "quality": 3,
            "upgrades": [
                "L1"
            ],
            "size": 3,
            "notes": null,
            "weapons": [
                {
                    "attacks": 4,
                    "count": 3,
                    "name": "Dual Hyper Swords",
                    "specialRules": [
                        {
                            "key": "rending",
                            "name": "Rending",
                            "rating": "",
                            "modify": false,
                            "label": "Rending"
                        }
                    ],
                    "ap": "1"
                }
            ],
            "specialRules": [
                {
                    "key": "ambush",
                    "name": "Ambush",
                    "rating": "",
                    "label": "Ambush"
                },
                {
                    "key": "robot",
                    "name": "Robot",
                    "rating": "",
                    "label": "Robot"
                },
                {
                    "id": 964,
                    "key": "self-repair",
                    "name": "Self-Repair",
                    "label": "Self-Repair"
                },
                {
                    "key": "strider",
                    "name": "Strider",
                    "rating": "",
                    "label": "Strider"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "3",
                    "label": "Tough(3)"
                }
            ]
        },
        {
            "type": "Doom Fighter",
            "name": "Kanoptech-Dominatorschreiter",
            "armyId": "4k5amkxoybdiqotm",
            "id": "xRIfeJR",
            "cost": 470,
            "defense": 2,
            "quality": 3,
            "upgrades": [],
            "size": 1,
            "notes": null,
            "weapons": [
                {
                    "attacks": 6,
                    "count": 1,
                    "name": "Doom Ray",
                    "range": 30,
                    "specialRules": [
                        {
                            "key": "lock-on",
                            "name": "Lock-On",
                            "rating": "",
                            "modify": false,
                            "label": "Lock-On"
                        }
                    ],
                    "ap": "4"
                },
                {
                    "attacks": 6,
                    "count": 1,
                    "name": "Twin Flux Cannon",
                    "range": 24,
                    "specialRules": [
                        {
                            "key": "flux",
                            "name": "Flux",
                            "rating": "",
                            "modify": false,
                            "label": "Flux"
                        }
                    ],
                    "ap": "1"
                }
            ],
            "specialRules": [
                {
                    "key": "aircraft",
                    "name": "Aircraft",
                    "rating": "",
                    "label": "Aircraft"
                },
                {
                    "key": "robot",
                    "name": "Robot",
                    "rating": "",
                    "label": "Robot"
                },
                {
                    "id": 964,
                    "key": "self-repair",
                    "name": "Self-Repair",
                    "label": "Self-Repair"
                },
                {
                    "key": "tough",
                    "name": "Tough",
                    "rating": "6",
                    "label": "Tough(6)"
                }
            ]
        },
        {
            "type": "Warriors",
            "name": "Necronkrieger",
            "armyId": "4k5amkxoybdiqotm",
            "id": "njiIkJi",
            "cost": 115,
            "defense": 4,
            "quality": 4,
            "upgrades": [
                "E1"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "Reaper Rifle",
                    "range": 18,
                    "specialRules": [],
                    "ap": "2"
                }
            ],
            "specialRules": [
                {
                    "id": 582,
                    "key": "robot",
                    "name": "Robot",
                    "label": "Robot"
                },
                {
                    "id": 964,
                    "key": "self-repair",
                    "name": "Self-Repair",
                    "label": "Self-Repair"
                },
                {
                    "key": "slow",
                    "name": "Slow",
                    "rating": "",
                    "label": "Slow"
                }
            ],
            "upgradeCost": [
                -5
            ]
        },
        {
            "type": "Warriors",
            "name": "Necronkrieger",
            "armyId": "4k5amkxoybdiqotm",
            "id": "njiIkJi",
            "cost": 115,
            "defense": 4,
            "quality": 4,
            "upgrades": [
                "E1"
            ],
            "size": 5,
            "notes": null,
            "weapons": [
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "CCW",
                    "specialRules": []
                },
                {
                    "attacks": 1,
                    "count": 5,
                    "name": "Reaper Rifle",
                    "range": 18,
                    "specialRules": [],
                    "ap": "2"
                }
            ],
            "specialRules": [
                {
                    "id": 582,
                    "key": "robot",
                    "name": "Robot",
                    "label": "Robot"
                },
                {
                    "id": 964,
                    "key": "self-repair",
                    "name": "Self-Repair",
                    "label": "Self-Repair"
                },
                {
                    "key": "slow",
                    "name": "Slow",
                    "rating": "",
                    "label": "Slow"
                }
            ],
            "upgradeCost": [
                -5
            ]
        }
    ]
}